CORP: Robotics industry NaN fix + better exports validation

Snarling opened the pull request:
1. Corrected the testing for player input on exports, now uses parentheses like the way the replacement is performed ingame.
2. Prevents exporting to the source city.
3. Prevents exporting to a city that does not have a warehouse for the target division. 
4. Provide better error messaging to player when an export fails.
    * Old error message, would display constantly during processing: 
![image](https://github.com/bitburner-official/bitburner-src/assets/84951833/bee6051a-0164-4f3f-8051-f13377827f7e)
    * New error message: Checks before setting export to prevent setting invalid amount: (outdated image, -IPROD is fine as an input)
![image](https://github.com/bitburner-official/bitburner-src/assets/84951833/05b6d11f-39b7-4bfc-8d08-d130dc813387)
5. Same validation is performed on gameload for saves older than 2.3.1 release.
    * Terminal error informs the player if any of their exports failed: 
![image](https://github.com/bitburner-official/bitburner-src/assets/84951833/17ee6755-4196-452e-b40e-7ccffb4e691c)
    * Console error shows full details ("sellAmt" is just the source division's name): 
![image](https://github.com/bitburner-official/bitburner-src/assets/84951833/74d7657f-43df-4577-9ef8-11a8ffdeaf7d)
6. Can no longer have the same material get exported from one warehouse to another multiple times through duplicate exports objects.
7. Canceling an export no longer requires the exact stored amount for the export.

Separate from exports, also fixed a NaN bug that could affect produced materials in Robotics industry (or any other industry that uses AI Cores as a required material)

Snarling committed:
Validate corp exports

d0sboots commented:
Wait, but the IPROD bug should be already fixed?

d0sboots commented:
I see. The issue is the the tests you're performing don't line up with how the expression is actually evaluated. (It's parenthesized when actually used).
We should do that generically instead of case-wise to be more safe.

Snarling commented:
Oh I guess I forgot that part was already addressed via a different fix, looks like I didn't backport it to 2.3 as a hotfix. I'll update the replacement to include parens (the updated validation is still needed to prevent exporting to same or nonexistent warehouse)

Snarling committed:
Update validation to use parens

Snarling committed:
More exports fixes

* Exports to a division are removed when selling that division
* Duplicate exports of the same material instance to the same warehouse are not allowed

Snarling committed:
Cancel exports without providing old amount

Now that exports are unique per source warehouse->destination warehouse, exports are uniquely identified just with division and city.

Snarling committed:
Fix robotics division NaN quality materials

Stop requiredMaterials from possibly going slightly negative due to floating point errors (clamped to 0)
Also in case they ever can still go negative, added a lower bound of 0 to AI cores in material quality calculation.
